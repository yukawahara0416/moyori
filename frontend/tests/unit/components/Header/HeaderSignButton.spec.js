import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import Component from '@/components/Header/HeaderSignButton.vue'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuetify)

let wrapper
let getters
let actions
let vuetify
let store

beforeEach(() => {
  getters = {
    headers: () => ({ uid: 'tester@example.com' }),
    dialogSign: () => true
  }

  actions = {
    dialogOn: jest.fn(),
    dialogOff: jest.fn()
  }

  store = new Vuex.Store({
    getters,
    actions
  })

  vuetify = new Vuetify()

  wrapper = mount(Component, {
    localVue,
    store,
    vuetify,
    stubs: ['sign-container']
  })
})

describe('getters', () => {
  it('headers', () => {
    expect(wrapper.vm.headers).toEqual(getters.headers())
  })
  it('dialogSign', () => {
    expect(wrapper.vm.dialogSign).toEqual(getters.dialogSign())
  })
})

// describe('computed', () => {
//   it('dialog', () => {})
// })

describe('v-on', () => {
  const app = document.createElement('div')
  app.setAttribute('data-app', true)
  document.body.append(app)

  it('dialogOn', () => {
    const event = jest.fn()
    wrapper.setMethods({ dialogOn: event })
    wrapper.find('.v-btn').trigger('click')
    expect(event).toHaveBeenCalledTimes(1)
  })
})

describe('actions', () => {
  it('dialogOn', () => {
    wrapper.vm.dialogOn()
    expect(actions.dialogOn).toHaveBeenCalled()
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
