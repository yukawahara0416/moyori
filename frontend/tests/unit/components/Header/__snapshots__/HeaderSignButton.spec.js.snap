// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<div
  class="v-dialog__container"
  role="dialog"
>
  <div
    class="my-2"
  >
    <button
      class="v-btn v-btn--depressed theme--light v-size--large primary"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        
        はじめる
      
      </span>
    </button>
  </div>
</div>
`;
